pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/generic/GenericAvroReader.java

```diff
@@ -66,7 +66,7 @@ public GenericAvroReader(Schema writerSchema, Schema readerSchema, byte[] schema
             this.reader = new GenericDatumReader<>(writerSchema, readerSchema);
         }
         this.byteArrayOutputStream = new ByteArrayOutputStream();
-        this.encoder = EncoderFactory.get().binaryEncoder(this.byteArrayOutputStream, encoder);
+        this.encoder = EncoderFactory.get().binaryEncoder(this.byteArrayOutputStream, null);
```

Comment:
Use null will avoid reusing the encode?

Reply:
It doesn't make any sense, the encoder in this time is always null. same below.